crates/sp-domains/src/fraud_proof.rs

```diff
@@ -345,3 +349,64 @@ pub struct ImproperTransactionSortitionProof {
     /// Hash of the bad receipt this fraud proof targeted
     pub bad_receipt_hash: ReceiptHash,
 }
+
+#[derive(Clone, Debug, Decode, Encode, Eq, PartialEq, TypeInfo)]
+pub struct TotalRewardsProof<BlockNumber, BlockHash> {
+    /// The id of the domain this fraud proof targeted
+    pub domain_id: DomainId,
+    /// Hash of the bad receipt this fraud proof targeted
+    pub bad_receipt_hash: ReceiptHash,
+    /// Number of the block
+    pub block_number: BlockNumber,
+    /// Hash of the domain block corresponding to `block_number`.
+    pub domain_block_hash: BlockHash,
+    /// Storage witness needed for verifying this proof.
+    pub storage_proof: StorageProof,
+}
+
+/// Represents an invalid bundle fraud proof.
+#[derive(Clone, Debug, Decode, Encode, Eq, PartialEq, TypeInfo)]
+pub enum InvalidBundleProof<BlockNumber, BlockHash> {
+    TotalRewards(TotalRewardsProof<BlockNumber, BlockHash>),
+}
```

Comment:
Although we consider a bundle invalid if it contains an invalid ER, it is strange to see `TotalRewardsProof` being wrapped by `InvalidBundleProof`, and if other fraud proofs of invalid ER fields follow the same structure then all the fraud proof (including invalid state transition as it is a fraud proof for `ER::execution_trace`) will be wrapped by `InvalidBundleProof`.

It may be out of focus of this PR, this also makes me wonder if we should hold "bundle is invalid if it contains an invalid ER" because:
- The operator will be slashed twice if it submits a bad ER but valid bundle (i.e. no invalid tx included), one for the fraudulent ER and one for being included in the next ER's `invalid_bundles` field
- The operator will be slashed once if it submits a valid ER but an invalid bundle (i.e. out of range tx) for being included in the next ER's `invalid_bundles` field
- The bundle itself doesn't have a strong connection to the ER they just submitted together, it seems no harm to continue to process the bundle even if the ER is invalid, if the bundle is indeed invalid then the operator will be slashed for that.

WDYT @vedhavyas

Reply:
Thinking on this more, I think it makes more sense to not consider them as invalid bundle.